### PR TITLE
feat: add autogenerated examples

### PR DESCRIPTION
**Pull request by armandobelardo**

### CODE REVIEW

File: packages/cli/ete-tests/src/tests/fdr/__snapshots__/fdr.test.ts.snap [hunk 1]
```diff
@@ -581,6 +610,44 @@ exports[`fdr {"name":"simple"} 1`] = `
                                 "type": "json",
                                 "value": "id-123"
                             }
+                        },
+                        {
+                            "path": "/",
+                            "pathParameters": {
+                                "movieId": "string"
```
Comment (dsinghvi) [inline packages/cli/ete-tests/src/tests/fdr/__snapshots__/fdr.test.ts.snap:617]: For OpenAPI when we do example generation, for strings we will copy the parameter name. Relevant [code](https://github.com/fern-api/fern/blob/9f1b2844bc4cdebc6e25e0bdcabf6acff94a4d12/packages/cli/openapi-parser/src/schema/examples/ExampleTypeFactory.ts#L487-L494)

Comment (dsinghvi) [reply packages/cli/ete-tests/src/tests/fdr/__snapshots__/fdr.test.ts.snap:617]: Maybe you can even use some of it

**Comment by dsinghvi** [comment]
If easy, it would be nice to share the same default example values for primitive types across the OAI importer and this. [Code](https://github.com/fern-api/fern/blob/9f1b2844bc4cdebc6e25e0bdcabf6acff94a4d12/packages/cli/openapi-parser/src/schema/examples/ExampleTypeFactory.ts#L486-L560)

File: packages/ir-sdk/fern/apis/ir-types-latest/definition/http.yml [hunk 2]
```diff
@@ -225,6 +225,12 @@ types:
 
   # examples
 
+  HttpEndpointExample:
+    discriminant: exampleType
+    union:
+      provided: ExampleEndpointCall
```
Comment (dsinghvi) [inline packages/ir-sdk/fern/apis/ir-types-latest/definition/http.yml:231]: nit: some docs here might be helpful or renaming the union key to `userProvided`

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+    TypeId,
+    TypeReference,
+    UndiscriminatedUnionTypeDeclaration,
+    UnionTypeDeclaration
+} from "@fern-api/ir-sdk";
+
+export class ExampleGenerator {
+    ir: IntermediateRepresentation;
+    private types: Map<TypeId, TypeDeclaration>;
+    private MAX_EXAMPLE_DEPTH = 2;
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:51]: The autogenerated examples here will also be used in docs if the user doesn't provide them. What we have been following for the OAI is: 
- Request (only contains required properties)
- Response (contains up to depth 2 of auto generated examples, unless the properties are required). 

Some [code](https://github.com/fern-api/fern-definition-registry/blob/d87477ba55d322b9e1cc3577ad5f79d7d45a0440/packages/fdr-sdk/src/converters/db/examples/generateHttpBodyExample.ts#L50-L74) in FDR

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+                };
+            default:
+                assertNever(primitiveType);
+        }
+    }
+
+    private generateExampleUnknown(): ExampleTypeReference {
+        return {
+            jsonExample: {},
+            shape: ExampleTypeReferenceShape.unknown({})
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:586]: for `unknown` the OpenAPI example generator creates a map `{ "key": "value" }` unless it is a header, query or path parameter. [code](https://github.com/fern-api/fern/blob/9f1b2844bc4cdebc6e25e0bdcabf6acff94a4d12/packages/cli/openapi-parser/src/schema/examples/ExampleTypeFactory.ts#L241-L249)

**Review by dsinghvi — APPROVED** [review]
@armandobelardo approving this so that it doesn't block your work. Couple things we need to do before merging: 
1. Update the migration so that you keep the `userProvided` examples around
2. Till we get the examples perfect can you make sure that the `ir-to-fdr` converter does **not** use the provided examples ([code](https://github.com/fern-api/fern/blob/9f1b2844bc4cdebc6e25e0bdcabf6acff94a4d12/packages/cli/register/src/ir-to-fdr-converter/convertPackage.ts#L96))

File: packages/ir-sdk/fern/apis/ir-types-latest/definition/http.yml (hunk 2 above)
Comment (armandobelardo) [reply packages/ir-sdk/fern/apis/ir-types-latest/definition/http.yml:231]: Updated!

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts (hunk 4 above)
Comment (armandobelardo) [reply packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:586]: Updated!

**Comment by armandobelardo** [comment]
closing in favor of https://github.com/fern-api/fern/pull/3114

**Pull request closed without merging**